src/sql/workbench/services/queryEditor/browser/queryEditorService.ts

```diff
@@ -47,7 +49,7 @@ export class QueryEditorService implements IQueryEditorService {
 				let docUri: URI = URI.from({ scheme: Schemas.untitled, path: filePath });
 
 				// Create a sql document pane with accoutrements
-				const fileInput = this._editorService.createInput({ forceUntitled: true, resource: docUri, mode: 'sql' }) as UntitledTextEditorInput;
+				const fileInput = this._editorService.createInput({ forceUntitled: true, resource: docUri, mode: this._resourceProviderService._providers[connectionProviderName].providerLanguageId }) as UntitledTextEditorInput;
```

Comment:
Same comment here. We cannot rely on shipped extensions to add languageId support. Map 'MSSQL' to 'sql', if languageId is undefined.